src/Nethermind/Nethermind.Evm/Tracing/CancellationTxTracer.cs

```diff
@@ -0,0 +1,394 @@
+            {
+                _innerTracer.ReportCodeChange(address, before, after);
+            }
+        }
+
+        public void ReportNonceChange(Address address, UInt256? before, UInt256? after)
+        {
+            _token.ThrowIfCancellationRequested();
+            if (_innerTracer.IsTracingState)
+            {
+                _innerTracer.ReportNonceChange(address, before, after);
+            }
+        }
+
+        public void ReportAccountRead(Address address)
+        {
+            _token.ThrowIfCancellationRequested();
+            if (_innerTracer.IsTracingState)
+            {
+                _innerTracer.ReportAccountRead(address);
+            }
+        }
+
+        public void ReportStorageChange(StorageCell storageCell, byte[] before, byte[] after)
+        {
+            _token.ThrowIfCancellationRequested();
+            if (_innerTracer.IsTracingState)
+            {
+                _innerTracer.ReportStorageChange(storageCell, before, after);
+            }
+        }
+        
+        public void ReportStorageRead(StorageCell storageCell)
+        {
+            _token.ThrowIfCancellationRequested();
+            if (_innerTracer.IsTracingState)
+            {
+                _innerTracer.ReportStorageRead(storageCell);
+            }
+        }
+
+        public void MarkAsSuccess(Address recipient, long gasSpent, byte[] output, LogEntry[] logs, Keccak stateRoot = null)
+        {
+            _token.ThrowIfCancellationRequested();
+            if (_innerTracer.IsTracingReceipt)
+            {
+                _innerTracer.MarkAsSuccess(recipient, gasSpent, output, logs, stateRoot);
+            }
+        }
+
+        public void MarkAsFailed(Address recipient, long gasSpent, byte[] output, string error, Keccak stateRoot = null)
+        {
+            _token.ThrowIfCancellationRequested();
+            if (_innerTracer.IsTracingReceipt)
+            {
+                _innerTracer.MarkAsFailed(recipient, gasSpent, output, error, stateRoot);
+            }
+        }
+
+        public void StartOperation(int depth, long gas, Instruction opcode, int pc)
+        {
+            _token.ThrowIfCancellationRequested();
+            if (_innerTracer.IsTracingInstructions)
+            {
+                _innerTracer.StartOperation(depth, gas, opcode, pc);
+            }
+        }
+
+        public void ReportOperationError(EvmExceptionType error)
+        {
+            _token.ThrowIfCancellationRequested();
+            if (_innerTracer.IsTracingInstructions)
+            {
+                _innerTracer.ReportOperationError(error);
+            }
+        }
+
+        public void ReportOperationRemainingGas(long gas)
+        {
+            _token.ThrowIfCancellationRequested();
+            if (_innerTracer.IsTracingInstructions)
+            {
+                _innerTracer.ReportOperationRemainingGas(gas);
+            }
+        }
+
+        public void SetOperationStack(List<string> stackTrace)
+        {
```

Comment:
This also extremely expensive - was it introduced to new operations?

Reply:
This is disabled by default. Only an option for complitnes.

Reply:
https://github.com/NethermindEth/nethermind/pull/2379/files#diff-9742ac85df13f5104064b9cf775ea804cc783bc16b9655d45c55774b13be19e6R552-R560